beacon_node/lighthouse_network/src/rpc/mod.rs

```diff
@@ -208,38 +219,59 @@ impl<Id: ReqId, E: EthSpec> RPC<Id, E> {
     pub fn send_response(
         &mut self,
         peer_id: PeerId,
-        id: (ConnectionId, SubstreamId),
-        _request_id: RequestId,
+        _id: (ConnectionId, SubstreamId),
+        request_id: RequestId,
         event: RpcResponse<E>,
     ) {
+        let Some(request) = self.active_inbound_requests.remove(&request_id) else {
+            error!(self.log, "Request not found in active_inbound_requests. Response not sent"; "peer_id" => %peer_id, "request_id" => ?request_id, "response" => %event);
+            return;
+        };
+
+        // Add the request back to active requests if the response is not a stream termination.
+        if request.r#type.protocol().terminator().is_some()
+            && !matches!(event, RpcResponse::StreamTermination(_))
+        {
+            self.active_inbound_requests
+                .insert(request_id, request.clone());
+        }
+
+        if let Some(response_limiter) = self.response_limiter.as_mut() {
+            if !response_limiter.allows(
+                peer_id,
+                request.r#type.protocol(),
+                request.connection_id,
+                request.substream_id,
+                event.clone(),
+            ) {
+                return;
+            }
+        }
+
         self.events.push(ToSwarm::NotifyHandler {
             peer_id,
-            handler: NotifyHandler::One(id.0),
-            event: RPCSend::Response(id.1, event),
+            handler: NotifyHandler::One(request.connection_id),
+            event: RPCSend::Response(request.substream_id, event),
         });
     }
 
     /// Submits an RPC request.
     ///
     /// The peer must be connected for this to succeed.
     pub fn send_request(&mut self, peer_id: PeerId, request_id: Id, req: RequestType<E>) {
-        let event = if let Some(self_limiter) = self.self_limiter.as_mut() {
-            match self_limiter.allows(peer_id, request_id, req) {
-                Ok(event) => event,
-                Err(_e) => {
-                    // Request is logged and queued internally in the self rate limiter.
-                    return;
-                }
+        match self
+            .outbound_request_limiter
```

Comment:
ok now I have understood the naming.
Honestly I don't think it makes sense for us to rate limit our responses, from what I read in the code we weren't previously right?
I read the [spec update](https://github.com/ethereum/consensus-specs/pull/3767/files) and what it mentions is:
>The responder MAY rate-limit chunks by withholding each chunk until capacity is available. The responder MUST NOT respond with an error or close the stream when rate limiting.

but that's not what we are doing here from what I read, we are applying the previous `rate_limiter` to our responses right? 
If we want back pressure and gauge the outbound throughput I'd rather do it the way [we did in gosipsub](https://github.com/libp2p/rust-libp2p/pull/5595)